_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,284 @@
+```html
+<div role="list">
+    <div role="listitem">List item 1</div>
+</div>
+```
+
+#### Passed Example 2
+
+Element with role `listitem` is contained within its required context role `list`, through the implicit role of `ul`.
+
+```html
+<ul>
+    <div role="listitem">List item 1</div>
+</ul>
+```
+
+#### Passed Example 3
+
+Element contained within its required context role even though it is not a direct child of the context role.
+
+```html
+<div role="list">
+    <div>
+         <div>
+             <div>
+                <div role="listitem">List item 1</div>
+             </div>
+         </div>
+    </div>
+</div>
+```
+
+#### Passed Example 4
+
+`aria-owns` used to give the target element the right context role.
+
+```html
+<div role="list" aria-owns="id1"></div>
+<div id="id1" role="listitem">List item 1</div>
+```
+
+#### Passed Example 5
+
+`aria-owns` trumps ownership by closest ancestor, giving the element with role of `listitem` the correct context role.
+
+```html
+<div role="list" aria-owns="id1">
+  <div role="tabpanel">
+    <div id="id1" role="listitem">List item 1</div>
+   </div>
+</div>
+```
+
+#### Passed Example 6
+
+Since implicit ownership can cross shadow boundaries, the element with the semantic role of `listitem` is contained within its required context role `list`.
+
+```html
+<div id="host" role="list">
+</div>
+
+<script>
+  const host = document.querySelector("#host");
+  const root = host.attachShadow({ mode: "open" });
+
+  root.innerHTML = `
+    <div role="listitem">List item 1</div>
+  `;
+</script>
+```
+#### Passed Example 7
+
+Element with role `listitem` is contained within its required context role `list`, expressed as an explicit role. Even though the element with role `listitem` is empty, it is still applicable to this rule.
+
+```html
+<div role="list">
+    <div role="listitem"></div>
+</div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+No context role.
+
+```html
+<div role="listitem">List item 1</div>
+```
+
+#### Failed Example 2
+
+Wrong context role.
+
+```html
+<div role="tablist">
+    <div role="listitem">List item 1</div>
+</div>
+```
+
+#### Failed Example 3
+
+Element not contained within its required context role.
+
+```html
+<div role="list"></div>
+    <div role="listitem">List item 1</div>
+```
+
+#### Failed Example 4
+
+Element with role `listitem` has a closer ancestor, that is included in the accessibility tree, than the role `list` that should have been its context role.
+
+```html
+<div role="list">
+    <div aria-label="menu">
+         <div role="listitem">List item 1</div>
+    </div>
+</div>
+```
+
+**Note:** This test case follows the definition of [owned by](#owned-by) used in this rule. If implemented differently, this definition could cause differences in outcome for this test case. 
+
+
+#### Failed Example 5
+
+Element with role `listitem` has a closer ancestor, that is included in the accessibility tree, than the role `list` that should have been its context role.
```

Comment:
Should we just say that `listitem` is owned by the `tabpanel`, because it is a closer ancestor?

Reply:
Fixed